docs/pages/database-access/rbac/configuring-auto-user-provisioning.mdx

```diff
@@ -0,0 +1,181 @@
+of shared database accounts for all users.
+
+<Admonition type="note" title="Supported databases">
+Currently, automatic user provisioning is only supported for self-hosted and
+RDS PostgreSQL databases.
+</Admonition>
+
+## Prerequisites
+
+- Teleport cluster with a configured [self-hosted PostgreSQL](../guides/postgres-self-hosted.mdx)
+  or [RDS PostgreSQL](../guides/rds.mdx) database.
+- Ability to connect to and create user accounts in the target database.
+
+## Step 1/3. Configure database admin
+
+Teleport should be able to connect to the database as a user that can create
+other users and assign them roles. We recommend creating a separate user
+designated specifically for Teleport automatic user provisioning. Let's call it
+`teleport-admin`.
+
+Teleport will use the same authentication mechanism when connecting as an admin
+user as for regular user connections: X.509 for self-hosted databases and AWS
+IAM for RDS. The admin user must have privileges within the database to create
+users and grant them privileges.
+
+<Tabs>
+<TabItem label="RDS PostgreSQL">
+The RDS PostgreSQL admin user must have the `rds_iam` role attached to allow IAM
+authentication:
+
+```sql
+CREATE USER "teleport-admin" login createrole;
+GRANT rds_iam TO "teleport-admin";
+```
+
+Note that the RDS database must have IAM authentication enabled.
+
+Refer to the [AWS documentation](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.DBAccounts.html) to make sure you are using the `rds_iam` role correctly.
+for more information.
+</TabItem>
+<TabItem label="Self-hosted PostgreSQL">
+The self-hosted PostgreSQL admin user must have X.509 authentication configured.
+
+```sql
+CREATE USER "teleport-admin" login createrole;
+```
+
+Note that the database must be configured to accept client certificate auth
+for the admin user by having the following entries in `pg_hba.conf`:
+
+```conf
+hostssl all             all             ::/0                    cert
+hostssl all             all             0.0.0.0/0               cert
+```
+
+Refer to the [self-hosted PostgreSQL guide](../guides/postgres-self-hosted.mdx#step-35-configure-your-postgresql-server)
+to ensure that your configuration is correct.
+</TabItem>
+</Tabs>
+
+Users created by Teleport will be placed in the `teleport-auto-users` group in
+the database, which will be created automatically if it doesn't exist.
+
+Teleport will not delete the automatically created user at the end of the session.
+Instead, the user will be stripped of all roles, updated with `nologin` trait
+and reactivated during the next connection.
+
+Next, enable the database admin on the Teleport database configuration:
+
+<Tabs>
+<TabItem label="Static config">
+```yaml
+db_service:
+  enabled: "yes"
+  databases:
+  - name: "example"
+    protocol: "postgres"
+    uri: "localhost:5432"
+    admin_user:
+      name: "teleport-admin"
+```
+</TabItem>
+<TabItem label="Dynamic resource">
+```yaml
```

Comment:
Should we include in this tab the appropriate `tctl edit` command to edit this resource (assuming one exists)?